### PR TITLE
[docs] Fix x-grid-data-generator dependencies

### LINKED ISSUE #1414: @material-ui/x-grid-data-generator not working
I've installed the XGrid and the x-grid-data-generator to test it following the example from the documentation, however, after implementing it, it breaks the application completely.

- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯
```
ERROR in C:/www/<app>/node_modules/@material-ui/x-grid-data-generator/dist/esm/index.js
Module not found: Error: Can't resolve '@material-ui/lab/Rating' in 'C:\www\<app>\node_modules\@material-ui\x-grid-data-generator\dist\esm'
```

## Expected Behavior 🤔
The application should render and show something like this
image

## Steps to Reproduce 🕹

Steps:

1. Install dependencies:
    "@material-ui/data-grid": "4.0.0-alpha.24",
    "@material-ui/icons": "~4.11.2",
    "@material-ui/x-grid": "4.0.0-alpha.24",
    "@material-ui/x-grid-data-generator": "4.0.0-alpha.24",
2. Implement the code from the documentation
3. Observe the error

The code from the documentation
```ts
import * as React from 'react';
import { XGrid } from '@material-ui/x-grid';
import { useDemoData } from '@material-ui/x-grid-data-generator';

export default function XGridDemo() {
  const { data } = useDemoData({
    dataSet: 'Commodity',
    rowLength: 100000,
  });

  return (
    <div style={{ height: 520, width: '100%' }}>
      <XGrid
        {...data}
        loading={data.rows.length === 0}
        rowHeight={38}
        checkboxSelection
      />
    </div>
  );
}
```

## Context 🔦
I am evaluating the XGrid for the project in my company.

## Your Environment 🌎

<details>
  <summary>`npx @material-ui/envinfo`</summary>
  
```
System:
    OS: Windows 10 10.0.17763
  Binaries:
    Node: 14.16.0 - ~\AppData\Local\Volta\tools\image\node\14.16.0\node.EXE
    Yarn: 1.22.10 - ~\AppData\Local\Volta\tools\image\yarn\1.22.10\bin\yarn.CMD
    npm: 6.14.11 - ~\AppData\Local\Volta\tools\image\node\14.16.0\npm.CMD
  Browsers:
    Chrome: 89.0.4389.114
    Edge: Not Found
  npmPackages:
    @emotion/styled:  10.0.27
    @material-ui/core: 4.11.3 => 4.11.3
    @material-ui/data-grid: 4.0.0-alpha.24 => 4.0.0-alpha.24
    @material-ui/icons: ~4.11.2 => 4.11.2
    @material-ui/styles:  4.11.3
    @material-ui/system:  4.11.3
    @material-ui/types:  5.1.0
    @material-ui/utils:  4.11.2
    @material-ui/x-grid: 4.0.0-alpha.24 => 4.0.0-alpha.24
    @material-ui/x-grid-data-generator: 4.0.0-alpha.24 => 4.0.0-alpha.24
    @material-ui/x-license:  4.0.0-alpha.21
    @types/react: 17.0.2 => 17.0.2
    react: 17.0.1 => 17.0.1
    react-dom: 17.0.1 => 17.0.1
    styled-components: 5.2.1 => 5.2.1
    typescript: 4.2.3 => 4.2.3
```
Browser: Chrome
</details>

### PR DESCRIPTION
**Pull request by ZeeshanTamboli**
Fixes #1414

### CODE REVIEW

File: packages/grid/x-grid-data-generator/package.json [hunk 1]
```diff
@@ -24,21 +24,19 @@
     "typescript": "tsc -p tsconfig.json"
   },
   "dependencies": {
+    "@material-ui/core": "^4.9.12 || ^5.0.0-alpha.22",
```
Comment (oliviertassinari) [inline packages/grid/x-grid-data-generator/package.json:27]: It should be a peer dependency. There is a singleton.

**Review by oliviertassinari — CHANGES_REQUESTED** [review]

File: packages/grid/x-grid-data-generator/package.json (hunk 1 above)
Comment (ZeeshanTamboli) [reply packages/grid/x-grid-data-generator/package.json]: Done. I just wanted to know. What do you mean by `singleton`? Does it mean there is no version conflict?

Comment (oliviertassinari) [reply packages/grid/x-grid-data-generator/package.json]: It means that it has little to no tolerance for duplication of the dependency in the bundle.

**Review by oliviertassinari — APPROVED** [review]

**Comment by oliviertassinari** [comment]
Giving it more thoughts, we might need to revert it for Material-UI v5 support. I'm not sure. Time will tell if it creates issues.

**Pull request merged**